docs/specifications/linking-and-embedding-specification.md

```diff
@@ -0,0 +1,364 @@
+- **Frontmatter**: Metadata section at the beginning of a note.
+  *See Also*: https://jekyllrb.com/docs/front-matter/
+- **Graph**: Graph structure that contains information about all the files in the workspace. Notes in the workspace are analyzed further e.g. for links and link targets.
+  *Also known as*: Index.
+- **Id**: Element attribute in the generated HTML that can be targeted with links / block embedding. Ids can also be used within the markdown files when defining anchors.
+- **Indexing**: Operation that goes through all the markdown files in the Foam workspace, finding all the links, block embeds, and link targets and finally linking these together in a Graph.
+- **Link Reference Definitions**: Information in a markdown file that converts wikilinks into regular markdown links.
+  *Also known as*: LinkRefDefs.
+  *See Also*: https://github.github.com/gfm/#link-reference-definitions
+- **Link**: A piece of text that points to a Target.
+- **Note**: A file containing text in markdown format.
+  *Also known as*: Foam Bubble, Document, Markdown file.
+- **Tag**: A piece of text that either acts as a link, or as an abstract categorization.
+- **Target**: An item (e.g. note, heading or an anchor) that can be pointed to with a link.
+- **Title**: The H1 heading of a note. A note can contain only a single title.
+- **Workspace**: The location (folder) where Foam Notes live in.
+
+## Targets
+
+The following items can be targetted via linking and / or embedding. The separation is explained in more detail in [Linking](#linking) and [Embedding](#embedding) chapters.
+
+### Notes
+
+Notes in Foam are individual markdown files. Notes can both reside in the Foam workspace root and any of its sub-folders. Both files and folders are encouraged to follow the kebab-case naming convention (e.g. `some/path-to/a-file.md`).
+
+### Headings
+
+Headings in Foam are denoted with markdown heading syntax, e.g. `## Heading on Second Level`. Each heading will have a corresponding id, a kebab-case version of the text (e.g. `heading-on-second-level`). In case there's multiple headings with the same text, their id is suffixed with a running number (e.g. `my-heading`, `my-heading-1`, `my-heading-2`).
+
+### Anchors
+
+User can add arbitrary anchors into a note with HTML-syntax: `<a name="my-anchor"></a>`. Since these names (identifiers) are written by users, they might end up being non-unique within a note. In these cases, an error is shown to the user. See the reasoning of using the name-attribute instead of the id-attribute from https://stackoverflow.com/a/7335259. In the future, better syntax for the anchors might be developed.
+
+### EBlocks
+
+While anchors define a single location in the document, eblocks define two; the beginning and the end of an eblock. The intent of a block is to specify content bounded by the begin and end markers. The end of an eblock can either be explicit or implicit.
```

Comment:
Conceptually, I do wonder, is there really a use case of explicitly defining EBlocks? It's a cool idea, but when do we need to refer to a fragment of text that's not already its own scope (paragraph, bullet, heading-ed section)?

I'm not certain. Let's make sure there's a case for it, since this requires advanced semantics.

Reply:
If someone writes in pure prose, they'll have multi-sentence paragraphs. Given what you guys are discussing about embeds and block references, I think its quite reasonable to expect the desire to link to just a sentence (or even word!). But if the difficulty of advanced semantics preclude this, that's perfectly understandable.